src/argilla/client/feedback/metrics/annotator_metrics.py

```diff
@@ -0,0 +1,373 @@
+
+    It contains two fields, `metric_name` and `result` with the value of the metric.
+    """
+
+    metric_name: str
+    result: Union[float, Dict[str, float], pd.DataFrame, List[pd.DataFrame]]
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class AnnotatorMetric:
+    def __init__(self, dataset: "FeedbackDataset", question_name: str) -> None:
+        """Initializes a `AnnotatorMetric` object to compute metrics on
+        a `FeedbackDataset` for a given question at the annotator level.
+
+        Args:
+            dataset: FeedbackDataset to compute the metrics.
+            question_name: Name of the question for which we want to analyse the analyze
+                the annotator responses vs the suggestions given.
+
+        Raises:
+            NotImplementedError: If the question type is not supported.
+        """
+        self._dataset = dataset
+        self._question_name = question_name
+        self._question_type = type(self._dataset.question_by_name(question_name))
+        if self._question_type in (MultiLabelQuestion, RankingQuestion):
+            raise NotImplementedError(f"No metrics are defined currently for {self._question_type.__name__}")
+        self._allowed_metrics = METRICS_PER_QUESTION[self._question_type]
+
+    def compute(self, metric_names: Union[str, List[str]], **kwargs) -> Dict[str, List[AnnotatorMetricResult]]:
+        """Computes the annotator metrics for the given question.
+
+        Args:
+            metric_names: name or list of names for the metrics to compute. i.e. `alpha`
+            kwargs: additional arguments to pass to the metric.
+
+        Raises:
+            ValueError: If the metric name is not supported for the given question.
+
+        Returns:
+            metrics: dict with the metrics computed for each annotator, where the
+                key corresponds to the user id and the values are a list with the
+                metric results.
+        """
+        if isinstance(metric_names, str):
+            metric_names = [metric_names]
+
+        if any([metric not in self._allowed_metrics for metric in metric_names]):
+            raise ValueError(
+                f"Metrics allowed for question {self._question_name}: {list(self._allowed_metrics.keys())}"
+            )
+
+        metric_classes = [(metric_name, self._allowed_metrics[metric_name]) for metric_name in metric_names]
+
+        responses_per_user, suggestions = get_responses_per_user(self._dataset, self._question_name)
+
+        metrics = defaultdict(list)
+        for user_id, responses in responses_per_user.items():
+            for metric_name, metric_cls in metric_classes:
+                metric = metric_cls(responses=responses, suggestions=suggestions)
+                result = metric.compute(**kwargs)
+                metrics[user_id].append(AnnotatorMetricResult(metric_name=metric_name, result=result))
+
+        return dict(metrics)
+
+
+class AnnotatorMetricBase(ABC):
```

Comment:
maybe this and all other Base modules might got in a base directory to adhere to the rest of the project structure